From b87ee16bf9b5a31fb3c776f57f5ed313732d0259 Mon Sep 17 00:00:00 2001
From: marionm <mathis.marion@grenoble-inp.org>
Date: Fri, 20 Aug 2021 11:49:25 +0200
Subject: [PATCH 2/4] Kernel 5.8 compatibility for i2c_new_device

The function `i2c_new_device` disappeared in kernel 5.8.
It is now required to use `i2c_new_client_device`.
Error check now requires using `IS_ERR` instead of just cheking for NULL.

Signed-off-by: Gwenhael GOAVEC <gwenhael.goavec@femto-st.fr>
Signed-off-by: Mathis MARION <mathis.marion@grenoble-inp.org>
---
 drivers/fmc/fmc-eeprom.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/drivers/fmc/fmc-eeprom.c b/drivers/fmc/fmc-eeprom.c
index 2bd2a6f..7ae4b11 100644
--- a/drivers/fmc/fmc-eeprom.c
+++ b/drivers/fmc/fmc-eeprom.c
@@ -153,8 +153,13 @@ static int __fmc_slot_eeprom_add(struct fmc_slot *slot,
 	}
 
 	info_l.addr = fmc_slot_i2c_address(info_l.addr, slot->ga);
+#if KERNEL_VERSION(5, 8, 0) <= LINUX_VERSION_CODE
+	slot->eeprom = i2c_new_client_device(slot->adapter, &info_l);
+	if (IS_ERR(slot->eeprom))
+#else
 	slot->eeprom = i2c_new_device(slot->adapter, &info_l);
 	if (!slot->eeprom)
+#endif
 		return -ENODEV;
 
 	err = sysfs_create_link(&slot->dev.kobj, &slot->eeprom->dev.kobj,
-- 
2.30.2

